packages/babel-helper-module-transforms/src/rewrite-live-references.ts

```diff
@@ -44,6 +44,24 @@ interface RewriteBindingInitVisitorState {
   scope: Scope;
 }
 
+function isInType(path) {
+  do {
+    switch (path.parent.type) {
+      case "TSTypeAnnotation":
+      case "TSTypeAliasDeclaration":
+      case "TypeAnnotation":
+      case "TypeAlias":
+        return true;
+      case "ExportSpecifier":
+        return path.parentPath.parent.exportKind === "type";
+      default:
+        if (path.parentPath.isStatement() || path.parentPath.isExpression()) {
```

Comment:
We should not exit the loop when `path.parentPath` is of alias typescript or flow. For example,

```ts
import { A } from "x";

var x = {} as A.b[0]
```